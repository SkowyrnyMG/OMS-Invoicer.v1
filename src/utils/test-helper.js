import React from 'react';
import { Router } from 'react-router-dom';
import { render, fireEvent } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { createMemoryHistory } from 'history';
import { theme } from 'themes/theme';

export const TestThemeProvider = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const renderWithRouter = (ui, { route = '/', ...renderOptions } = {}) => {
  const history = createMemoryHistory({ initialEntries: [route] });
  const utils = render(<Router history={history}>{ui}</Router>, renderOptions);
  return {
    ...utils,
    history,
  };
};

const leftClick = { button: 0 };

export { leftClick, renderWithRouter, fireEvent };
