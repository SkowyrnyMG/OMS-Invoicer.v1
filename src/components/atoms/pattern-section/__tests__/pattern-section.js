import React from 'react';

import { snapShotTest } from 'utils/tests/test-helper';
import PatternSection from '../pattern-section';

describe('PatternSection', () => {
  test('should display in the document with correct styles', () => {
    snapShotTest(
      <PatternSection patternUrl='https://somelinkwithpattern.com'>
        <div />
      </PatternSection>,
    );
  });
});
