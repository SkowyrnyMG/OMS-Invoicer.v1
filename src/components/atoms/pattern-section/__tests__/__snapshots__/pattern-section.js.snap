// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatternSection should display in the document with correct styles 1`] = `
.c0 {
  grid-column: -1 / 1;
  display: grid;
  grid-template-columns: 5rem [start] 1fr [end] 5rem;
  grid-column-gap: 3%;
  padding: 20rem 0;
  position: relative;
  background-image: url(https://somelinkwithpattern.com);
  background-size: cover;
  background-position: center;
}

.c0 > * {
  grid-column: start / end;
}

.c0 {
  padding: 10rem 0;
}

<section
  class="c0"
>
  <div />
</section>
`;
